hooks/terraform_docs.sh

```diff
@@ -142,10 +142,16 @@ function terraform_docs {
     local tf_docs_formatter="md"
 
   else
-
+    have_config_flag=true
     local config_file=${args#*--config}
     config_file=${config_file#*=}
-    config_file=${config_file% *}
+    # If there are more parameters after config path, trim until --
+    if [[ $config_file == *" --"* ]]; then
+      config_file=${config_file%% --*}
+    fi
+    # Trim trailing whitespace but preserve internal spaces
+
+    config_file="${config_file%"${config_file##*[![:space:]]}"}" args=${args/--config=$config_file/}
```

Comment:
```suggestion
    # Enable extended pattern matching operators
    shopt -qp extglob || EXTGLOB_IS_NOT_SET=true && shopt -s extglob
    # Trim any args before the `--config` arg value
    local config_file=${args##*--config(+([[:space:]])|=)}
    # Trim any trailing args (if any)
    config_file="${config_file%%+([[:space:]])--*}"
    # Trim `--config` arg and its value from original args
    # as we will pass `--config` separately to allow whitespaces
    # in its value
    args=${args/--config=$config_file*([[:space:]])}
    # Restore state of `extglob` if we changed it
    [[ $EXTGLOB_IS_NOT_SET ]] && shopt -u extglob
```

Hmm, the `config_file=${config_file#*=}` was doing a lot of oddity as it trimmed spaces if space was used to separate `--config` from value 😕 So I replaced it.

@tarfu @MaxymVlasov Would you guys have a chance to test and verify this code please? 🙏🏻

Reply:
ℹ️ I resolved my other related threads in favour of this commit suggestion.

Reply:
I tested a bunch of different combinations of test cases, though I still don't have any real-life use case to validate it in field conditions 🤷🏻 
One case that I have no idea how to cover in Bash w/o a whole huge monstrous kludge is when config file path contains space-dash-dash (like `/my/p --ath/to/file`) which is possible but wth 🤪

Reply:
Ah, yeah, tested to work with bash from ~era of dinosaurs~ macOS:
```shell
GNU bash, version 3.2.0(5)-beta (x86_64-unknown-linux-gnu)
Copyright (C) 2005 Free Software Foundation, Inc.
```

Reply:
For me, your change result in:

```
grep: unrecognized option `--config=/Users/tobias.breitwieser/tmp/path space/path space/repo/.terraform-docs.yml'
usage: grep [-abcdDEFGHhIiJLlMmnOopqRSsUVvwXxZz] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
grep: unrecognized option `--config=/Users/tobias.breitwieser/tmp/path space/path space/repo/.terraform-docs.yml'
usage: grep [-abcdDEFGHhIiJLlMmnOopqRSsUVvwXxZz] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
rm: illegal option -- -
usage: rm [-f | -i] [-dIPRrvWx] file ...
       unlink [--] file
```

I'm on a mac.
Bash Version:
```
GNU bash, version 3.2.57(1)-release (arm64-apple-darwin24)
Copyright (C) 2007 Free Software Foundation, Inc.
```
grep version:
`grep (BSD grep, GNU compatible) 2.6.0-FreeBSD`

Reply:
I committed the changes to make suggestions easier to do ^^.

Reply:
Let me check.